exporter/awss3exporter/factory.go

```diff
@@ -57,6 +58,10 @@ func createMetricsExporter(ctx context.Context,
 		return nil, err
 	}
 
+	if config.(*Config).MarshalerName == SumoIC {
```

Comment:
Please add a `Validate()` method to the Config struct and run this validation there so it can be picked up by validation tools.

Reply:
I'm not sure I understand how it should be done.
The correct condition for this error is:
`the marshaler name is SumoIC AND the exporter has been defined in a pipeline for metrics or traces`
It's not possible to check the pipeline inside the `Validate` method, is it? Because after adding this method and running it in `createMetricsExporter` and `createTracesExporter`, the test `TestMarshallerName` in `config_test.go` fails.

Reply:
OK, that makes sense, sorry.